network/p2p/connection/connector.go

```diff
@@ -132,28 +132,32 @@ func (l *Libp2pConnector) pruneAllConnectionsExcept(peerIDs peer.IDSlice) {
 		}
 
 		peerInfo := l.host.Network().Peerstore().PeerInfo(peerID)
-		log := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
-		if l.host.ConnManager().IsProtected(peerID, "") {
-			log.Trace().Msg("skipping pruning since connection is protected")
-			continue // connection is protected (stream or connection in progress), skip pruning
-		}
+		lg := l.log.With().Str("remote_peer", peerInfo.String()).Logger()
+
+		// log the protected status of the connection
+		protected := l.host.ConnManager().IsProtected(peerID, "")
+		lg = lg.With().Bool("protected", protected).Logger()
 
-		// retain the connection if there is a Flow One-to-One stream on that connection
-		// (we do not want to sever a connection with on going direct one-to-one traffic)
+		// log if any stream is open on this connection.
 		flowStream := p2putils.FlowStream(conn)
 		if flowStream != nil {
-			log.Info().
-				Str("stream_protocol", string(flowStream.Protocol())).
-				Msg("skipping connection pruning with peer due to one-to-one stream")
-			continue // flow stream found, skip pruning
+			lg = lg.With().Str("flow_stream", string(flowStream.Protocol())).Logger()
 		}
 
 		// close the connection with the peer if it is not part of the current fanout
 		err := l.host.Network().ClosePeer(peerID)
 		if err != nil {
-			log.Error().Err(err).Msg("failed to disconnect from peer")
+			// logging with suspicious level as failure to disconnect from a peer can be a security issue.
+			// e.g., failure to disconnect from a malicious peer can lead to a DoS attack.
+			lg.Error().
+				Bool(logging.KeySuspicious, true).
+				Err(err).Msg("failed to disconnect from peer")
 		} else {
```

Comment:
nit: remove the else block

Reply:
We also need to log when we disconnect from a peer (part of adding observability into the networking layer). The other block does that happy path logging of disconnection. However, to address this comment, I simplified the structure of the code:
https://github.com/onflow/flow-go/pull/3459/commits/929315a6e0297dfec57c29a45dfad3b63c8544ab